### ISSUE TITLE
Multi input model issue

### ISSUE THREAD

**Issue by nbrosson** [keras-team/keras#12793]
Hello,

I would like to do a multi input model for a classification problem. However, I have two input: One with 50 000 rows, and one with 10 000 rows. 

The labels are the same, but, obviously, with different sizes. Here is what I did:

```
left_branch = Input((8, ))
left_branch_dense = Dense(5, activation = 'tanh')(left_branch)
        
right_branch = Input((44350, ))
right_branch_reduc = Dense(100, activation = 'tanh')(right_branch)
right_branch_dense = Dense(5, activation = 'tanh')(right_branch_reduc)
merged = Concatenate()([left_branch_dense, right_branch_dense])
output_layer = Dense(2, activation = 'sigmoid')(merged)
      
model = Model(inputs=[left_branch, right_branch], outputs=output_layer)
model.compile(loss='sparse_categorical_crossentropy', optimizer='adam', metrics=['accuracy'])

model.fit([batch_1, batch_2], [labels_bgp_1,labels_bgp_2])
```

However, I have the following error: 
**Error when checking model target: the list of Numpy arrays that you are passing to your model is not the size the model expected. Expected to see 1 array(s), but instead got the following list of 2 arrays: [array([[1.],
       [0.],
       [1.],
       ...,
       [0.],
       [0.],
       [0.]]), array([[0.],
       [1.],
       [1.],
       ...,
       [0.],
       [1.],
       [0.]])]...**

Looks like the model expect only one label array. What should I do?? 

Thanks a lot !!

**Comment by jvishnuvardhan** [comment]
Please fill the issue [template](https://github.com/keras-team/keras/issues/new?template=a--tensorflow-backend-users.md). Could you update them if they are relevant in your case, or leave them as N/A? Along with the template, please provide as many details as possible to find the root cause of the issue. It would be great if you can provide a small code to reproduce the error. Thanks!
